dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AliasableResourceInterface.java

```diff
@@ -49,27 +56,75 @@
      */
     T getAndCheckResourceByAlias(String alias);
 
-    default T addAliases(User user, Long id, String aliases) {
-        T c = getAndCheckResource(user, id);
-        Set<String> oldAliases = c.getAliases().keySet();
-        Set<String> newAliases = Sets.newHashSet(Arrays.stream(aliases.split(",")).map(String::trim).toArray(String[]::new));
+    /**
+     * Check that aliases do not contain invalid prefixes
+     * if the user adding them is not an admin or curator
+     * @param aliases a Set of alias strings
+     * @param user user authenticated to issue a DOI for the workflow
+     * @param blockAliasesWithParticualarFormat block creation of an alias with a particular format
+     * @return the alias as a string
+     */
+    static void checkAliases(Set<String>  aliases, User user, boolean blockAliasesWithParticualarFormat) {
+        // Gather up any aliases that contain invalid prefixes
+        List<String> invalidAliases = new ArrayList<>();
+        if (!user.isCurator() && !user.getIsAdmin()) {
```

Comment:
`!user.isCurator() && !user.getIsAdmin()` is in both if branches here. 
It seems that this could be simplified as 
```
if (user.isCurator() || user.getIsAdmin()) {return;}
```
i.e. it looks like the way this is written, curators and admins can do anything

Reply:
done (simplified)